app/controllers/api/v1/comments_controller.rb

```diff
@@ -39,13 +56,37 @@ def report
         if report.save
           head :no_content
         else
-          render(
-            json:   {error: I18n.t("report.status.failed")},
-            status: :internal_server_error
-          )
+          render json: I18n.t("api.endpoint_errors.comments.duplicate_report"), status: :conflict
+        end
+      end
+
+      private
+
+      def comment_and_post_validate(post_guid, comment_guid)
+        if !comment_exists(comment_guid)
+          render json: I18n.t("api.endpoint_errors.comments.not_found"), status: :not_found
+          false
+        elsif !comment_is_for_post(post_guid, comment_guid)
+          render json: I18n.t("api.endpoint_errors.comments.not_found"), status: :not_found
+          false
+        else
+          true
         end
       end
 
+      def comment_is_for_post(post_guid, comment_guid)
```

Comment:
I was wondering: what is the best place for this function, controller or service?

Reply:
I thought about that when I was writing this.  This isn't being used except in our API because it's protecting from a condition that is not possible on the Rails side so I left it in our controller instead.